### PR TITLE
Display camera preview directly in TakePictureFragment

### PR DESCRIPTION
**Pull request by stblr**

### CODE REVIEW

File: app/src/main/java/com/github/wnder/TakePictureFragment.java [hunk 1]
```diff
@@ -83,80 +93,143 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
...
+            AlertBuilder.okAlert(getString(R.string.guest_not_allowed), getString(R.string.guest_no_upload), view.getContext())
+                    .show();
+        } else {
+            ActivityResultLauncher<String> requestPermissionLauncher =
+                    registerForActivityResult(new ActivityResultContracts.RequestPermission(), isGranted -> {
+                        if (isGranted) {
+                            initializeCameraPreview(view);
+                        }
+                    });
+
```
Comment (rkuenzi-epfl) [inline app/src/main/java/com/github/wnder/TakePictureFragment.java:114]: This could probably go in the else statement right before launching?

**Review by rkuenzi-epfl — CHANGES_REQUESTED** [review]
You should add some Javadoc to your function and add few comments in general because I really think what you do won't seem obvious to the others that will have to work on this soon. 
Otherwise it's ok.
For the tests I guess we will have to do without them on cirrus, but please leave them as comment so that if we think we broke something with your code we can at least test it locally.

**Review by rkuenzi-epfl — APPROVED** [review]
For me it's ok, but the bug on cirrus is even weirder than expected I think?
I though the camera couldn't be opened on cirrus but from the test you committed here it seems there is no issues with the camera opening but it prevents you from using the buttons once it's opened which is really strange. So I guess we will have to do with these partial tests.

**Review by amattei777 — APPROVED** [review]
LGFM

**Pull request merged**